UI: Refactor navbar to align with the update of core ui

ChengYanJin opened the pull request:
**Component**: ui

**Context**: 
As there are breaking changes in core-ui(release version 0.3.0), we will need to update the version of core-ui and refactor the Navbar at the same time. (covered metalk8s-UI and example-solution UI)

**Summary**:

- Removed toggleVisible, applications, help, user and languages from Navbar.
- Add `rightActions` with the list of actions in Navbar.
- Implement the language the logic.

**Acceptance criteria**: 
We won't see the dropdown in Navbar if there is no item. 

---

Closes: #2085

ChengYanJin committed:
ui: add necesssary library for core-ui

Add package: vega, vega-lite and vega-embed

Refs: #2085

ChengYanJin committed:
ui: refectoring the navbar

remove the ToggleVisible, languages, applications, help and user

Refs: #2085

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality commented on ui/src/containers/Layout.js:
```diff
@@ -143,14 +128,44 @@ const Layout = props => {
     },
   ];
 
+  const filterLanguage = languages.find(lang => lang.name !== language);
```
you should use `filter` instead of `find` here. In our case, we have 2 languages, that's why it works.

carlito-scality commented on ui/src/containers/Layout.js:
```diff
@@ -143,14 +128,44 @@ const Layout = props => {
+      type: 'dropdown',
+      text: language,
+      icon: <i className="fas fa-globe" />,
+      items: filterLanguage === null ? null : [filterLanguage],
```
`items: filterLanguage` is enough if you use `filter` above.

carlito-scality commented on ui/src/containers/Layout.js:
```diff
@@ -143,14 +128,44 @@ const Layout = props => {
+    {
+      type: 'dropdown',
+      icon: <i className="fas fa-th" />,
+      items: applications === null ? null : [applications],
```
`items: applications` is enough because applications is an array. if `applications` is null, you should not add it to `rightActions`

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @carlito-scality

ChengYanJin committed:
ui/example-solution: refactor the navbar in solution example ui.

- Remove toggleVisible, application, helps and user in Navbar
- Remove application item in example-solution-UI as we don't swtich back to Metalk8s
- Add rightActions

Refs: #2085

carlito-scality commented on ui/src/containers/Layout.js:
```diff
@@ -143,14 +128,44 @@ const Layout = props => {
+    {
+      type: 'dropdown',
+      icon: <i className="fas fa-th" />,
+      items: applications,
```
you should check if there is applications not before adding this `rightActions`

carlito-scality commented on examples/metalk8s-solution-example/ui/src/containers/Layout.js:
```diff
@@ -87,13 +73,36 @@ const Layout = props => {
+    {
+      type: 'dropdown',
+      icon: <i className="fas fa-th" />,
+      items: applications
```
same here, you should check if there is applications not before adding this rightActions

ChengYanJin replied:
I am wondering in solution-example UI do we need applications item in `Navbar`?

carlito-scality commented on examples/metalk8s-solution-example/ui/src/containers/Layout.js:
```diff
@@ -49,20 +49,6 @@ const Layout = props => {
 
   const applications = [];
```
you should update the core-ui in package.json of metalk8s-solution-example as well

ChengYanJin committed:
ui/example-solution: add necessary library for example solution ui

- Add package: vega, vega-lite and vega-embed
- Update pockage-lock.json. The explaintion to the optional:true
  (https://stackoverflow.com/a/55373854)

Refs: #2085

ChengYanJin committed:
ui: increase the limit the maxAssetSize

We import vega that reach out the limit the bundle size.

Refs: #2085

carlito-scality commented on ui/src/containers/Layout.js:
```diff
@@ -143,14 +127,46 @@ const Layout = props => {
+          type: 'dropdown',
+          icon: <i className="fas fa-th" />,
+          items: applications,
+        },
```
please move this check outside this array and insert 
```
{
  type: 'dropdown',
  icon: <i className="fas fa-th" />,
  items: applications,
 }
```
only if `applications && applications.length`

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.5`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.5`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`



Please check the status of the associated issue None.

Goodbye chengyanjin.
